st2api/st2api/controllers/resource.py

```diff
@@ -355,7 +365,28 @@ def _validate_exclude_fields(self, exclude_fields):
         return exclude_fields
 
 
-class ContentPackResourceController(ResourceController):
+class BeseResourceIsolationHandlerMixin(object):
+    """
+    Isolate resources for users except system_user.
+    """
+    @staticmethod
+    def resource_model_filter(model, instances, requester_user=None,
+                              offset=0, eop=0, **kwargs):
+        result = []
+        for instance in instances[offset:eop]:
+            item = model.from_model(instance, **kwargs)
+            if not cfg.CONF.rbac.permission_isolation:
+                result.append(item)
+            elif requester_user.name == cfg.CONF.system_user.user:
```

Comment:
I think we should also allow users with admin permission to view all the resources.

In fact, that's how we already handle it right now for user scoped datastore items - regular users can only see items scoped to themselves, but admins can see all the datastore items or the items scoped to a particular user.

This also brings up another question - to be consistent with datastore API endpoint, we should probably allow admin user to see items for a particular user (e.g. by using `?user=foo` or `?scope_user=foo` query parameter or similar).

Reply:
Edit: Never mind. Looking at the code below, it looks like that is already supported - https://github.com/StackStorm/st2/pull/4013/files#diff-9a7b5df7b28bc1b332b8a777a5930f9cR104

Can you please confirm.

Reply:
@Kami  I should have discussed this with you. I assumed all items could only be seen by platform manager(system_admin). There is not a specific permission type for view and execute(create, update, get, delete) items belongs to a particular user, so I extend this permission limit on `admin`. Other common user has only permission of viewing.